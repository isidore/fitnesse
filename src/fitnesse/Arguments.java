// Copyright (C) 2003-2009 by Object Mentor, Inc. All rights reserved.
// Released under the terms of the CPL Common Public License version 1.0.
package fitnesse;

public class Arguments {
  public static final String DEFAULT_PATH = ".";
  public static final String DEFAULT_ROOT = "FitNesseRoot";
  public static final int DEFAULT_PORT = 80;
  public static final int DEFAULT_VERSION_DAYS = 14;

  private String rootPath = DEFAULT_PATH;
  private int port = DEFAULT_PORT;
  private String rootDirectory = DEFAULT_ROOT;
  private String logDirectory;
  private boolean omitUpdate = false;
  private int daysTillVersionsExpire = DEFAULT_VERSION_DAYS;
  private String userpass;
  private boolean installOnly;

  public String getRootPath() {
    return rootPath;
  }

  public void setRootPath(String rootPath) {
    this.rootPath = rootPath;
  }

  public int getPort() {
    return port;
  }

  public void setPort(String port) {
    this.port = Integer.parseInt(port);
  }

  public String getRootDirectory() {
    return rootDirectory;
  }

  public void setRootDirectory(String rootDirectory) {
    this.rootDirectory = rootDirectory;
  }

  public String getLogDirectory() {
    return logDirectory;
  }

  public void setLogDirectory(String logDirectory) {
    this.logDirectory = logDirectory;
  }

  public void setOmitUpdates(boolean omitUpdates) {
    this.omitUpdate = omitUpdates;
  }

  public boolean isOmittingUpdates() {
    return omitUpdate;
  }

  public void setUserpass(String userpass) {
    this.userpass = userpass;
  }

  public String getUserpass() {
    if (userpass == null || userpass.length() == 0)
      return null;
    else
      return userpass;
  }

  public int getDaysTillVersionsExpire() {
    return daysTillVersionsExpire;
  }

  public void setDaysTillVersionsExpire(String daysTillVersionsExpire) {
    this.daysTillVersionsExpire = Integer.parseInt(daysTillVersionsExpire);
  }

  public boolean isInstallOnly() {
    return installOnly;
  }

  public void setInstallOnly(boolean installOnly) {
    this.installOnly = installOnly;
  }

  @Override
  public String toString() {
    return String.format("Port: %s \r\n" + "Root Path: %s\r\n"
        + "RootDirectory: %s\r\n" + "LogDirectory: %s\r\n" + "%s"
        + "Days Till Version Expire: %s \r\n" + "UserPassword File: %s\r\n"
        + "%s"

    , port, rootPath, rootDirectory, logDirectory,
        (omitUpdate ? "Omitting Updates \r\n" : ""),
        getDaysTillVersionsExpire(), getUserpass(),
        isInstallOnly() ? "Install Only \r\n" : "");
  }
}
