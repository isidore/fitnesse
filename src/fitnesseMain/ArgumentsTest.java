// Copyright (C) 2003-2009 by Object Mentor, Inc. All rights reserved.
// Released under the terms of the CPL Common Public License version 1.0.
package fitnesseMain;

import junit.framework.TestCase;

import org.approvaltests.Approvals;

import fitnesse.Arguments;

public class ArgumentsTest extends TestCase {

  public void testSimpleCommandline() throws Exception {
    approveArguments(null);
  }

  private void approveArguments(String a) throws Exception {
    String[] split = a == null ? new String[0] : a.split(" ");
    Arguments args = FitNesseMain.parseCommandLine(split);
    Approvals.approve(args.toString());
  }

  public void testArgumentsAlternates() throws Exception {
    approveArguments("-p 123 -d MyWd -r MyRoot -l LogDir -e 322 -o -a userpass.txt -i");
  }

  public void testAllArguments() throws Exception {
    approveArguments("-p 81 -d directory -r root -l myLogDirectory -o -e 22");
  }

  public void testNotOmitUpdates() throws Exception {
    approveArguments("-p 81 -d directory -r root -l myLogDirectory");
  }

  public void testBadArgument() throws Exception {
    assertNull(FitNesseMain.parseCommandLine((new String[] { "-x" })));
  }
}
