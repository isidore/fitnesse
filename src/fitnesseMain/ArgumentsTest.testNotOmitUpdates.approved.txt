Port: 81 
Root Path: directory
RootDirectory: root
LogDirectory: myLogDirectory
Days Till Version Expire: 14 
UserPassword File: null
